File: src/main/java/io/jsql/hazelcast/test/AtomicLongMain.java
package io.jsql.hazelcast.test;
import com.hazelcast.core.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by 长宏 on 2017/5/3 0003.
 */
public class AtomicLongMain  {
    static Logger logger = LoggerFactory.getLogger("me");
    public static void main(String[] args) {
        HazelcastInstance hz = Hazelcast.newHazelcastInstance();
        IAtomicLong counter = hz.getAtomicLong("counter");
        counter.addAndGet(3); // value is now 3
        counter.alter(new MultiplyByTwo());//value is now 6
        IdGenerator idGenerator = hz.getIdGenerator("myid");

        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        logger.info(idGenerator.newId() + "");
        System.out.println("counter: "+counter.get());
    }

    public static class MultiplyByTwo implements IFunction<Long,Long> {
        @Override
        public Long apply(Long input) {
            return input*2;
        }
    }

}

File: src/main/java/io/jsql/hazelcast/test/AtomicLongMain.kt
/*
 * Java-based distributed database like Mysql
 */

package io.jsql.hazelcast.test

import com.hazelcast.core.Hazelcast
import com.hazelcast.core.IFunction
import org.slf4j.LoggerFactory

/**
 * Created by 长宏 on 2017/5/3 0003.
 */
object AtomicLongMain {
    internal var logger = LoggerFactory.getLogger("me")
    @JvmStatic
    fun main(args: Array<String>) {
        val hz = Hazelcast.newHazelcastInstance()
        val counter = hz.getAtomicLong("counter")
        counter.addAndGet(3) // value is now 3
        counter.alter(MultiplyByTwo())//value is now 6
        val idGenerator = hz.getIdGenerator("myid")

        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        logger.info(idGenerator.newId().toString() + "")
        println("counter: " + counter.get())
    }

    class MultiplyByTwo : IFunction<Long, Long> {
        override fun apply(input: Long?): Long? {
            return input!! * 2
        }
    }

}
